compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -520,9 +520,14 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    node->setRegister(source);
 
 
-   //printf("Node %p is first child\n", firstChild); fflush(stdout);
-   //traceMsg(comp, "Node %p is first child\n", firstChild);
-   cg->decReferenceCount(firstChild);
+   if (shouldGenerateGuardedLoadsForCompressedLoad)
+      {
+      cg->decReferenceCount(firstChild->getFirstChild());
```

Comment:
I don't think in general it's safe to skip evaluating a node.  It might be possible in some cases, but the circumstances have to be well controlled (and documented that this is what you're doing and why).  I'll defer to an optimizer expert to make the determination as to whether its safe in this case.

I don't think you need a recursive decReferenceCount here because the firstChild was evaluated, which should have done the node decrementing as part of that process.